qcodes/instrument_drivers/rohde_schwarz/RTO1000.py

```diff
@@ -0,0 +1,560 @@
+# All manual references are to R&S RTO Digital Oscilloscope User Manual
+# for firmware 3.65, 2017
+
+import numpy as np
+import warnings
+from distutils.version import LooseVersion
+
+from qcodes import Instrument
+from qcodes.instrument.visa import VisaInstrument
+from qcodes.instrument.channel import InstrumentChannel
+from qcodes.utils import validators as vals
+from qcodes.instrument.parameter import ArrayParameter
+
+
+class ScopeTrace(ArrayParameter):
+
+    def __init__(self, name: str, instrument: InstrumentChannel,
+                 channum: int) -> None:
+        """
+        The ScopeTrace parameter is attached to a channel of the oscilloscope.
+
+        For now, we only support reading out the entire trace.
+        """
+        super().__init__(name=name,
+                         shape=(1,),
+                         label='Voltage',  # TODO: Is this sometimes dbm?
+                         unit='V',
+                         setpoint_names=('Time',),
+                         setpoint_labels=('Time',),
+                         setpoint_units=('s',),
+                         docstring='Holds scope trace')
+
+        self.channel = instrument
+        self.channum = channum
+
+    def prepare_trace(self) -> None:
+        """
+        Prepare the scope for returning data, calculate the setpoints
+        """
+
+        # we must have this nesting, since the parameter high_definition_state
+        # only exists if HD
+        if self.channel._parent.HD:
+            if self.channel._parent.high_definition_state() == 'ON':
+                dataformat = 'INT,16'
+            else:
+                dataformat = 'INT,8'
+        else:
+            dataformat = 'INT,8'
+
+        self.channel._parent.dataformat(dataformat)
+        # ensure little-endianess
+        self.channel._parent.write('FORMat:BORder LSBFirst')
```

Comment:
Would we gain any speed from merging these writes into one? and is it even possible

Reply:
Yes, we probably would. But there is an underlying assumption that this speed-up is small compared to the time scale of everything else. It would only happen once BEFORE a loop, for instance, and thus never really be a true bottleneck.